sorted_db/db.go

```diff
@@ -186,6 +204,42 @@ func (db *DB) Search(needle []byte) []byte {
 	return nil
 }
 
+// RangeMatch uses binary searches to look for startNeedle and (if not nil)
+// endNeedle. Returns all full match lines that fall between startNeedle and
+// endNeedle, inclusive. startNeedle and endNeedle should already both end
+// with the record separator.
+func (db *DB) RangeMatch(startNeedle []byte, endNeedle []byte) []byte {
+	db.RLock()
+
+	if db.size <= 0 {
+		panic("DB not Mapped")
+	}
+	startRecord := db.findFirstRecordAfterMatch(startNeedle, true)
+	if startRecord < 0 || startRecord == db.size {
+		db.RUnlock()
+		return nil
+	}
+	previous := lastIndexByte(db.data, startRecord, db.LineEnding)
+	// eat the line ending or (if there is no line ending) move to the start
```

Comment:
The precedent idiom for this action in the file is:

```go
    if previous == -1 {
      previous = 0
    } else {
      previous++ // eat the line ending
    }
```
It took me a couple reads to grok what was happening in this line. As written, it makes me think there was a conditional check that I'm not seeing.

However, this is the more succinct way (or even `startIndex := previous++`) to accomplish the same thing. I think clearer documentation would indicate we are setting the index to the first non-line-ending byte or to the start of the file.